src/shared/utils/vis.ts

```diff
@@ -315,54 +315,27 @@ export const getMainColumnName = (
   return ''
 }
 
-const getS2CellID = (table: Table, index: number): string => {
-  const column = table.getColumn('s2_cell_id')
-  if (!column) {
+const getColumnValue = (table: Table, field: string) => {
+  const fieldColumn = table.getColumn('_field')
+
+  if (!fieldColumn) {
     throw new Error(
-      'Cannot retrieve s2_cell_id column - table does not conform to required structure of Table type'
+      'Cannot retrieve _field column - table does not conform to required structure of Table type'
     )
   }
+  const index = fieldColumn.findIndex(val => val === field)
+  const valueColumn = table.getColumn('_value')
+  const value = valueColumn[index]
 
-  const value = column[index]
-  if (typeof value !== 'string') {
-    throw new Error('invalid s2_cell_id column value - value must be a string')
-  }
   return value
 }
-/* 
-   The geo precision table value calculated below is utilized by S2-geometry, a library which enhances the accuracy and efficiency of point / range coordinates on a map by accounting 
-   for the semi-spherical nature of the earth rather than attempting to plot against a 2d rendering. The cellId value and the precision table are used to ascertain 
-   the cell id at the particular level of specificity we are looking for, and that is then passed to S2 to retrieve the lat/lon value at that id. 
-*/
-
-const getPrecisionTrimmingTableValue = (): bigint[] => {
-  const precisionTable = [BigInt(1)]
-  for (let i = 1; i <= HEX_DIGIT_PRECISION; i++) {
-    precisionTable[i] = precisionTable[i - 1] * BigInt(HEX_DIGIT_PRECISION)
-  }
-  return precisionTable
-}
-
-export const getGeoCoordinates = (
-  table: Table,
-  index: number
-): {lon: number; lat: number} | null => {
-  const cellId = getS2CellID(table, index)
-
-  if (cellId.length > HEX_DIGIT_PRECISION) {
-    throw new Error(
-      'invalid cellId length - value must not be longer than the defined hex digit precision'
-    )
-  }
-
-  const fixed =
-    BigInt('0x' + cellId) *
-    getPrecisionTrimmingTableValue()[HEX_DIGIT_PRECISION - cellId.length]
 
-  const geoCoordinateValue = S2.idToLatLng(fixed.toString())
+export const getGeoCoordinates = (table: Table) => {
+  const latCoordinate = getColumnValue(table, 'lat')
+  const lonCoordinate = getColumnValue(table, 'lon')
 
   return {
-    lat: geoCoordinateValue.lat,
-    lon: geoCoordinateValue.lng,
+    lat: parseInt(latCoordinate.toString()),
+    lon: parseInt(lonCoordinate.toString()),
```

Comment:
```suggestion
    lat: parseInt(latCoordinate.toString(), 10),
    lon: parseInt(lonCoordinate.toString(), 10),
```

you generally always want to include a radix when calling `parseInt` because different browsers might implement it differently without one.

https://stackoverflow.com/questions/6611824/why-do-we-need-to-use-radix-parameter-when-calling-parseint

Reply:
good idea, I updated it!